axon-server-connector/src/main/java/org/axonframework/axonserver/connector/InstructionResultPublisher.java

```diff
@@ -0,0 +1,25 @@
+package org.axonframework.axonserver.connector;
+
+/**
+ * Interface that encapsulates the instruction result publication functionality.
+ *
+ * @author Sara Pellegrini
+ * @since 4.4
+ */
+public interface InstructionResultPublisher {
+
+    /**
+     * Notifies to Axon Server a successful execution of the specified instruction.
+     *
+     * @param instructionId the identifier of the instruction that has been successfully processed.
+     */
+    void publishSuccessFor(String instructionId);
+
+    /**
+     * Notifies to Axon Server a failure during the execution of the specified instruction.
+     *
+     * @param instructionId the identifier of the instruction.
+     * @param error         the error happened during the instruction execution.
+     */
+    void publishFailureFor(String instructionId, Throwable error);
```

Comment:
Not all failures are represented by a `Throwable`. Having an overload with a String representing a cause could be beneficial. What do you think?